rust/src/main.rs

```diff
@@ -11,19 +11,48 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-use std::io;
 
 #[macro_use]
 extern crate log;
 extern crate chrono;
 extern crate fern;
 
+extern crate dirs;
+
 extern crate xi_core_lib;
 extern crate xi_rpc;
 
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+
 use xi_core_lib::XiCore;
 use xi_rpc::RpcLoop;
 
+const XI_DIRECTORY: &str = "xi-core";
+const XI_LOG_FILE: &str = "xi-core.log";
+
+fn get_logging_directory() -> Result<PathBuf, io::Error> {
+    match dirs::data_local_dir() {
+        Some(mut log_dir) => {
+            log_dir.push(XI_DIRECTORY);
+            Ok(log_dir)
+        }
+        None => Err(io::Error::new(
+            io::ErrorKind::NotFound,
+            "The dir library was not able to find a directory for this platform",
```

Comment:
Nit: the wording here is quite vague, something like "No standard logging directory known for this platform" would be more descriptive I think? Mentioning the library is not necessary either.

That aside, nice job!

Reply:
You're right. I'd thought I'd at least added "logging directory"; good spot.
Also yea I guess knowing the library is a bit useless. Very easy to check where the log came from & see the library from there.
Changes pushed.